tests/cupy_tests/sorting_tests/test_sort.py

```diff
@@ -650,6 +650,22 @@ def test_argpartition_multi_dim(self, xp, dtype):
                 a[rows, cols, idx[:, :, kth + 1:]]).all()
         return idx[:, :, kth:kth + 1]
 
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_argpartition_multi_dim_kernel(self, xp, dtype):
+        a = testing.shaped_random((3, 3, 256), xp, dtype, 100)
+        kth = 2
+        idx = self.argpartition(a, kth, axis=-1)
+
+        rows = [[[0]], [[1]], [[2]]]
+        cols = [[[0], [1], [2]]]
+
+        assert (a[rows, cols, idx[:, :, :kth]] <
+                a[rows, cols, idx[:, :, kth:kth + 1]]).all()
+        assert (a[rows, cols, idx[:, :, kth:kth + 1]] <
+                a[rows, cols, idx[:, :, kth + 1:]]).all()
```

Comment:
We're more likely to draw the same numbers in 256 samples out of 100 which happen to be partitioned at kth compared to `test_argpartition_multi_dim`.

```suggestion
        assert (a[rows, cols, idx[:, :, :kth]] <=
                a[rows, cols, idx[:, :, kth:kth + 1]]).all()
        assert (a[rows, cols, idx[:, :, kth:kth + 1]] <=
                a[rows, cols, idx[:, :, kth + 1:]]).all()
```